code/z_adventurezones/lavamoon.dm

```diff
@@ -774,57 +785,61 @@ var/global/iomoon_blowout_state = 0 //0: Hasn't occurred, 1: Moon is irradiated
 				src.health -= I.force * 0.5
 
 
-			if (src.health <= 0 && active != -1)
-				src.set_dir(2)
-				src.active = -1
+			if (src.health <= 0 && functional != -1)
+				src.icon_state = "bot_spawner_dead"
+				src.functional = 0
 				src.visible_message("<span class='alert'>[src] shuts down. Forever.</span>")
 				return
 
 
 
 		proc/spawn_bot()
-			if (active || (max_bots  < 1))
+			if (makingbot || !functional || (max_bots  < 1))
 				return -1
 
-			active = 1
-			src.set_dir(1)
+			makingbot = 1
+			src.icon_state = "bot_spawner_start"
+			SPAWN(0.7 SECONDS)
+				src.icon_state = "bot_spawner_active"
 			src.visible_message("<span class='alert'>[src] begins to whirr ominously!</span>")
 			SPAWN(2 SECONDS)
 				if (health <= 0)
-					set_dir(2)
+					src.icon_state = "bot_spawner_dead"
 					return
-				src.set_dir(4)
 				if(prob(50)) //cheese reduction
 					src.visible_message("<span class='alert'>[src] produces a terrifying vibration!</span>")
 					for(var/atom/A in orange(3, src))
 						if(!(ismob(A) || iscritter(A))) //only target inanimate objects mostly
 							A.ex_act(1)
 				sleep(1 SECOND)
 				if (health <= 0)
-					set_dir(2)
+					src.icon_state = "bot_spawner_dead"
 					return
+				src.icon_state = "bot_spawner_finish"
+				sleep (0.4 SECONDS)
 				if (prob(80))
 					new /mob/living/critter/robotic/repairbot (src.loc)
 				else
 					new /mob/living/critter/robotic/repairbot/security (src.loc)
 				max_bots--
 
 				src.visible_message("<span class='alert'>[src] plunks out a robot! Oh dear!</span>")
-				active = 0
-				set_dir(2)
+				makingbot = 0
 
 			return
 
 	core
 		name = "mechanism core"
 		desc = "An enormous artifact of some sort. You feel uncomfortable just being near it."
-		icon = 'icons/misc/worlds.dmi'
-		icon_state = "powercore_core_dead"
-		layer = 4.5 // TODO LAYER
+		icon = 'icons/effects/96x96.dmi'
+		icon_state = "powercore_core"
+		layer = 10 // TODO LAYER
+		plane = PLANE_NOSHADOW_ABOVE
 
-		var/active = 0
+		var/IOCORE_ONLINE = 0
+		var/IOCORE_DEAD = 0
```

Comment:
This could be a single status var using defines.

Before the core you would add:

#define IOCORE_INACTIVE 0
#define IOCORE_ACTIVE  1
#define IOCORE_DEAD  2

Then you would have a `var/status = IOCORE_INACTIVE`
You would then check and assign `status` as appropriate, eg `if (src.status == IOCORE_ACTIVE )`

The defines have to be removed after the blockof code they are used in (so after the end of core)

#undef IOCORE_INACTIVE 
#undef IOCORE_ACTIVE  
#undef IOCORE_DEAD  

You can find an implementation like this already on the core, STATE_DEFAULT, STATE_MARKER_OUT and STATE_RECHARGING are used to control the state for spawning or zapping.